### PR TITLE
Add WebsocketServer exception handler

### PR DESCRIPTION
**Pull request by davidbrochart**
The ability to auto-restart is only supported when using the start/stop API for now, not the context manager.
It does so by catching exceptions and re-creating a new task group.

### CODE REVIEW

File: pycrdt_websocket/websocket_server.py [hunk 1]
```diff
@@ -198,11 +209,20 @@ async def start(
...
+            while True:
+                try:
+                    async with create_task_group() as self._task_group:
+                        if not self.started.is_set():
+                            task_status.started()
+                            self.started.set()
+                        # wait forever
+                        self._task_group.start_soon(Event().wait)
+                    break
+                except Exception as e:
```
Comment (jzhang20133) [inline pycrdt_websocket/websocket_server.py:221]: When the original task group fail, all its tasks are cancelled, including all the existing rooms's broadcast_update tasks, do we need to find a way to recover them? 
Is it possible to have one task group per room so one room throws exception in broadcast_update task won't also tear down other room's  broadcast_update task?

Comment (jzhang20133) [reply pycrdt_websocket/websocket_server.py:221]: Could we also add auto_restart for yroom's task group as well?

Comment (Zsailer) [reply pycrdt_websocket/websocket_server.py:221]: I think we should explore recovering each individual room too, but maybe let's leave this for a follow-up PR? 

At the _very minimum_ we need recovery at the top layer.

**Review by Zsailer — APPROVED** [review]

**Comment by Zsailer** [comment]
This looks good to me @davidbrochart.

**Comment by Zsailer** [comment]
I like this a lot, @davidbrochart. Great work!

**Comment by Zsailer** [comment]
this looks good, @davidbrochart. I propose we merge here and work on getting #30 to follow this pattern. If we notice any changes needed here in that PR, we can fix there. 

I'll work with @jzhang20133 to get these two aligned.

**Comment by davidbrochart** [comment]
Sounds good.

**Pull request merged**